ai-agent/src/main/scala/wvlet/ai/agent/LLMAgent.scala

```diff
@@ -90,4 +90,21 @@ case class LLMAgent(
   /** Remove any tool choice configuration. */
   def noToolChoice: LLMAgent = this.withModelConfig(_.noToolChoice)
 
+  /**
+    * Create a chat session with tool execution support.
+    *
+    * @param runner
+    *   The agent runner to use
+    * @param toolExecutor
+    *   Optional tool executor for automatic tool execution
+    * @return
+    *   A tool-enabled chat session
+    */
+  def newToolEnabledSession(
```

Comment:
Using tools is already optional in the parameter, so name this method as `newSession`